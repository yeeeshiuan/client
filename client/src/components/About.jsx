import React, { Component } from 'react';

class About extends Component {
    constructor() {
        super();
        this.state = {
        };
    }
    render() {
        return (
                <div>
                    <p>This is my App.(About)</p>
                </div>
        )
    }
}

export default About;
