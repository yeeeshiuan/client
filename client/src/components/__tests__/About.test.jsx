import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import About from '../About';


test('About renders without crashing', () => {
  const wrapper = shallow(<About/>);
});

test('About contain a p tag', () => {
  const wrapper = shallow(<About />);
  const p = wrapper.find('p');
  expect(p.length).toBe(1);
  expect(p.get(0).props.children).toBe('This is my App.(About)');
});

test('About renders a snapshot properly', () => {
  const tree = renderer.create(<About/>).toJSON();
  expect(tree).toMatchSnapshot();
});
