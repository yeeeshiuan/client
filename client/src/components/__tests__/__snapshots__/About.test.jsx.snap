// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders a snapshot properly 1`] = `
<div>
  <p>
    This is my App.(About)
  </p>
</div>
`;
